### ISSUE TITLE
Relax promise-function-async for short arrow functions

### ISSUE THREAD

**Issue by bajtos** [palantir/tslint#2637]
### Feature request

- __TSLint version__:  4.5.1
- __TypeScript version__:  2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
async function foo(x) {
  return x+1;
}

const bar = () => foo(1);
```

with `tslint.json` configuration:

```json
{
  "$schema": "http://json.schemastore.org/tslint",
  "rules": {
    "promise-function-async": true
  }
}
```

#### Actual behavior

```
ERROR: x.ts[5, 13]: functions that return promises must be async
```

#### Expected behavior

I am proposing to relax the rule and allow a single statement lamda function that's only delegating to another promise-returning function as a valid code.

My reasoning is that making a function `async` comes with overhead, both in performance but also the size of the generated javascript. 

Consider the following code snippet:

```ts
async function foo() {
  return 1;
}

const bar = () => foo();

const baz = async () => foo();
```

and compare the generated output:

```js
// target: es5

var bar = function () { return foo(); };
var baz = function () { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
    return [2 /*return*/, foo()];
}); }); };

// target: es6

const bar = () => foo();
const baz = () => __awaiter(this, void 0, void 0, function* () { return foo(); });
```
